### ISSUE TITLE
Significant change of behavior

### ISSUE THREAD

**Issue by MartinquaXD** [foundry-rs/foundry#8035]
### Component

Anvil

### Have you ensured that all of these are up to date?

- [ ] Foundry
- [ ] Foundryup

### Operating System

None

### Describe the bug

Our CI suddenly started [failing](https://github.com/cowprotocol/services/actions/runs/9314666954/job/25703978714) on a test that has never been unstable. I traced it back to `anvil` and went over the recent commits to see which one introduced the change. From what I can tell it was https://github.com/foundry-rs/foundry/pull/7984 which caused the regression.
Unfortunately I don't have a minimal way to reproduce the issue but I can give you the test that started failing in case you want to debug with that:
1. check out https://github.com/cowprotocol/services
2. run `docker compose up` from the root directory
3. make sure that the respective `anvil` version can be found (by putting it in the `PATH` variable or copying the version into `~/.foundry/bin/anvil`)
4. `cargo test -p e2e -- --ignored local_node_combined`

The tests makes a few assertions on the balances of a few addresses but I don't know how relevant that is specifically given that the linked PR is only related to gas prices.

**Comment by mattsse** [comment]
what's the issue that's causing

> 2024-06-02T16:03:27.825Z ERROR observe::tracing: thread 'protocol_fee::local_node_combined_protocol_fees' panicked at crates/e2e/tests/e2e/protocol_fee.rs:366:5:

https://github.com/cowprotocol/services/blob/84fe9c6d423128fa1f6feab43f9a18b54589e5fa/crates/e2e/tests/e2e/protocol_fee.rs#L366-L369

this pr changed priority fee estimation from a constant to dynamic.

happy to tmp revert that change if you can find out what's broken

**Comment by MartinquaXD** [comment]
I looked a bit more into this issue. My process was basically to partially revert the commit to have the code functions for versions available and then assert that the gas prices returned by each version had to be identical. Not sure if this should be the case but this seems reasonable to me.
What I noticed is that the new code path returns a gas price of 0 whereas the old code returned a gas price of 1GWEI for us.
This might be due to the specific configuration we use. (`--base-fee=0 --gas-price=1`). My assumption is that the gas price should not go below the price configure via CLI but with the new PR this is no longer the case.
My dirty hack which ended up passing my assertion was to do `gas_price.max(configured_price)` to ensure that we not end up with a zero gas price.

**Comment by MartinquaXD** [comment]
We now fixed our e2e tests and I gave a few more details on the issue in https://github.com/cowprotocol/services/pull/2771. To me the recent commit still seems like a breaking change since the computed gas price is different across the 2 implementations but at least we are not blocked on getting this resolved in `anvil` itself anymore.

**Comment by mattsse** [comment]
> returns a gas price of 0

ah see how this can happen, this is an oversight, I consider this a bug
